src/stdfblib/events/E_RESTART.cpp

```diff
@@ -89,7 +89,9 @@ EMGMResponse FORTE_E_RESTART::changeFBExecutionState(EMGMCommandType paCommand){
   if(EMGMResponse::Ready == eRetVal){
     switch(paCommand){
       case EMGMCommandType::Start:
-        mEventToSend = (scmEventSTOPID == mEventToSend) ? scmEventWARMID : scmEventCOLDID;
+        if (scmEventWARMID != mEventToSend) {
```

Comment:
I am not sure if this is the right approach here.
Just to verify, the current implemented behavior here is, that WARM is sent if the FB has been STOPPED earlier, and COLD is only sent if the E_RESTART FB has just been created, but never used before.

Now, how I understand the IEC 61131, a PLC can either get the command to COLD or WARM start anytime during its operation. Is this somehow handled then by the IDE? Like if a COLD start command is used, then everything is teared down on FORTE and then rebuild? Or is this handled somewhere else in FORTE if a COLD restart is requested by the user?

Reply:
I think i misunterstood something here before..If i unterstand it correctly there is no such a thing like a warm or cold command, the restart function block just determines the event to be sent from the transitions of the state machine. In order to trigger RESET and setIntitialValues we first need to send KILL, RESET and START. I was wrong here because, if i read the standard correctly then this should also trigger a COLD restart event. the warm restart event should only occur if the block has been stopped but not killed. Is this correct?

Reply:
With standard you mean the IEC 61499 I assume? If yes, I need to check on that first.
i am asking, because some part of the desired functionality is also based on the IEC 61131 (I think part 1), but perhaps I am understanding the intend here wrong.